pkgs/applications/editors/vim/plugins/overrides.nix

```diff
@@ -1558,21 +1559,37 @@ assertNoAdditions {
     ];
   };
 
-  kulala-nvim = super.kulala-nvim.overrideAttrs {
-    dependencies = with self; [
-      nvim-treesitter
-      nvim-treesitter-parsers.http
-    ];
-    buildInputs = [ curl ];
-    postPatch = ''
-      substituteInPlace lua/kulala/config/defaults.lua \
-        --replace-fail 'curl_path = "curl"' 'curl_path = "${lib.getExe curl}"'
-    '';
-    nvimSkipModules = [
-      # Requires some extra work to get CLI working in nixpkgs
-      "cli.kulala_cli"
-    ];
-  };
+  kulala-nvim = super.kulala-nvim.overrideAttrs (
+    old:
+    let
+      kulala-http-grammar = neovimUtils.grammarToPlugin (
+        tree-sitter.buildGrammar {
+          inherit (old) version src meta;
+          language = "kulala_http";
+          location = "lua/tree-sitter";
+          generate = false;
+        }
+      );
```

Comment:
This looks like what I had in my personal flake, completely forgot about upstreaming this thanks for handling it.